pkg/resources/compactor/servicemonitor.go

```diff
@@ -25,6 +25,7 @@ func (c *Compactor) serviceMonitor() (runtime.Object, reconciler.DesiredState, e
 	if c.ObjectStore.Spec.Compactor != nil && c.ObjectStore.Spec.Compactor.Metrics.ServiceMonitor {
 		metrics := c.ObjectStore.Spec.Compactor.Metrics
 		meta := c.getMeta()
+		meta.Name = "thanos-store-" + meta.Name
```

Comment:
maybe this should/could be fixed in the getMeta function to match with the other resource that compactor creates

Reply:
I think this was a bad merge/rebase from my side.

Reply:
I mean I didn't want to go into naming fixes in this PR for other parts of thanos :)

Reply:
fair enough!